Source/Order.swift

```diff
@@ -54,16 +89,37 @@ final public class Order: NSObject, Codable {
                   smallest decimal unit of the currency (e.g. 3999 for $39.99).
         - currencyCode: The ISO 4217 currency code (default is USD).
      */
+    @available(*, deprecated, message: "Use init(id:currencyCode:purchaseDate:customerOrderId:lineItems:customer:) instead")
     @objc public init(id: String, amount: Int64 = 0, currencyCode: String = "USD") {
         self.id = id
         self.amount = amount
         self.currencyCode = currencyCode
     }
+    
+    /**
+     Initializes an order object with the passed parameters.
+     
+     - Parameters:
+     - id: The order identifier (required).
+     - currencyCode: The ISO 4217 currency code (default is USD).
+     - purchaseDate: The date of the purchase for the order.
+     - lineItems: A list of the line item details that comprise the order.
+     */
+    @objc public init(id: String, currencyCode: String = "USD", purchaseDate: Date, lineItems: [LineItem]) {
```

Comment:
I think we should remove currencyCode so ObjC consumers are not required to pass a value that would otherwise have a default of "USD"